src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_INDEX = 0.75;
+    private int capacityArray = 16;
+    private MyNode<K,V> [] arrayBaskets;
+    private int size;
+
+    public MyHashMap() {
+        arrayBaskets = new MyNode [capacityArray];
+        size = 0;
+    }
+
+    private void addToList(MyNode<K, V> newNode, int index) {
+        MyNode<K, V> currentNode = arrayBaskets[index];
+        MyNode<K, V> memCurrentNode = currentNode;
+        while (currentNode != null) {
+            if (currentNode.getKey() == null && newNode.getKey() == null
+                    || currentNode.getKey() != null
+                    && currentNode.getKey().equals(newNode.getKey())) {
+                currentNode.setValue(newNode.getValue());
+                size--;
+                return;
+            }
+            memCurrentNode = currentNode;
+            currentNode = currentNode.getNext();
+        }
+        memCurrentNode.setNext(newNode);
+    }
+
+    private void resize() {
+        size = 0;
+        capacityArray = capacityArray << 1;
+        MyNode<K, V> [] memArrayBaskets = arrayBaskets;
+        arrayBaskets = new MyNode[capacityArray];
+        for (int i = 0; i < memArrayBaskets.length; i++) {
+            if (memArrayBaskets[i] != null) {
+                put(new MyNode<K,V>(memArrayBaskets[i].getKey(), memArrayBaskets[i].getValue()));
+                MyNode<K,V> current = memArrayBaskets[i].getNext();
+                while (current != null) {
+                    put(new MyNode<K,V>(current.getKey(), current.getValue()));
+                    current = current.getNext();
+                }
+            }
+        }
+    }
+
+    private void put(MyNode<K,V> newNode) {
+        int indexBasket = Math.abs(newNode.getHashKey() % capacityArray);
```

Comment:
you can do something like that and it would be better to create a separate method for this
```suggestion
        int indexBasket = newNode.key == null ? 0 : Math.abs(newNode.key.hashCode() % capacityArray);
```

Reply:
здесь не вижу смысла исправлять, т.к. считаю, что это непринципиально и мой вариант мне нравится больше, т.к. эту проверку уже делаю в конструкторе